VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts

```diff
@@ -145,4 +150,23 @@ export class AppComponent implements OnInit, OnDestroy {
             })
         );
     }
+
+    @HostListener('window:beforeunload', ['$event'])
+    async beforeunloadHandler($event: any) {
+        $event.preventDefault();
+        await this.participantStatusUpdateService.postParticipantStatus(EventType.ParticipantNotSignedIn);
+        $event.returnValue = 'save';
+    }
+
+    refreshPageParticipant() {
+        this.subscriptions.add(this.router.events.subscribe(async (event) => {
```

Comment:
It looks like the app component is subscribing to the same observable twice. Once here and once in `scrollToTop`.

There should be a function per each bit of logic that needs to execute on a `NavigationEnd` event. This will make it easier to track what behaviour is executing against this observable

Reply:
If I subscribe to catch refresh from router event  in ngOnInit , it does not work. In constructor it does work. But scrolling should be in ngOnInit as it depends on DOM element.